rsl/rsl.go

```diff
@@ -0,0 +1,80 @@
+package rsl
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/adityasaky/gittuf/common"
+	"github.com/adityasaky/gittuf/internal/gitinterface"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+const RSLRef = "refs/gittuf/reference-state-log"
+
+/*
+InitializeRSLNamespace creates a git ref for the reference state log. Initially,
+the entry has a zero hash.
+*/
+func InitializeRSLNamespace() error {
```

Comment:
For all the funcs in this package, you can probably drop the `RSL` from the name, since it's already in the package. https://google.github.io/styleguide/go/decisions#repetition

e.g.

- rsl.InitializeRSLNamespace -> rsl.InitializeNamespace
- rsl.AddRSLEntry -> rsl.AddEntry
- ...